### PR TITLE
Enqueue Server collector

### PR DESCRIPTION
**Pull request by stefanotorresi**
This PR implements the SAP Enqueue Server collector via the `EnqGetStatistics` SAPControl method.

It also adds the first draft of a metrics document.

### CODE REVIEW

File: collector/enqueue_server/enqueue_server.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+)
+
+func NewCollector(webService sapcontrol.WebService) (*EnqueueServerCollector, error) {
+
+	c := &EnqueueServerCollector{
+		collector.NewDefaultCollector("enqueue_server"),
+		webService,
+	}
+
+	c.SetDescriptor("owner_now", "TBD", nil)
```
Comment (MalloZup) [inline collector/enqueue_server/enqueue_server.go]: `TBD`?

Comment (stefanotorresi) [reply collector/enqueue_server/enqueue_server.go]: To Be Defined. Honestly, I don't know where to get the descriptions yet!

File: doc/metrics.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+### `sap_enqueue_server_arguments_state`
+
+#### Example
+
+```
+# TYPE sap_enqueue_server_arguments_state gauge
+sap_enqueue_server_arguments_state 2
+```
+
+### `sap_enqueue_server_backup_requests`
```
Comment (MalloZup) [inline doc/metrics.md]: we might open a follow-up issue to document better this

File: doc/sap_host_exporter.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# sample config
+port: "9680"
+addres: "0.0.0.0"
+log-level: "info"
+sap-control-url: ""
+sap-control-user: ""
+sap-control-pass: ""
```
Comment (MalloZup) [inline doc/sap_host_exporter.yaml]: perhaps is better `sap-control-password`. I think `pass` is not really full clear

File: doc/metrics.md (hunk 2 above)
Comment (stefanotorresi) [reply doc/metrics.md]: yeah, I need to dig through the SAP documentation further, to understand how to describe all these metrics.

File: doc/sap_host_exporter.yaml (hunk 3 above)
Comment (stefanotorresi) [reply doc/sap_host_exporter.yaml]: fair enough!

File: collector/enqueue_server/enqueue_server.go (hunk 1 above)
Comment (MalloZup) [reply collector/enqueue_server/enqueue_server.go]: ok please open a issue on github and we can follow-up. Also put the link of the issue as comment in code

**Pull request merged**